src/anchoredinterval.jl

```diff
@@ -223,6 +223,10 @@ end
 
 ##### EQUALITY #####
 
+function Base.:(==)(a::AnchoredInterval{P, T}, b::AnchoredInterval{P, T}) where {P, T}
```

Comment:
We should implement `isequal` here as well.

Reply:
We can safely fall back to `isequal(::AbstractInterval, ::AbstractInterval)` here. I'll add some tests though to make sure we have the proper behaviour.